lib/features/mailbox_dashboard/presentation/extensions/validate_saas_premium_available_extension.dart

```diff
@@ -0,0 +1,53 @@
+import 'package:flutter/material.dart';
+import 'package:tmail_ui_user/features/home/domain/extensions/session_extensions.dart';
+import 'package:tmail_ui_user/features/mailbox_dashboard/presentation/controller/mailbox_dashboard_controller.dart';
+import 'package:tmail_ui_user/features/paywall/domain/model/paywall_url_pattern.dart';
+import 'package:tmail_ui_user/features/paywall/domain/usecases/get_paywall_url_interactor.dart';
+import 'package:tmail_ui_user/main/localizations/app_localizations.dart';
+import 'package:tmail_ui_user/main/routes/route_navigation.dart';
+import 'package:tmail_ui_user/main/utils/app_utils.dart';
+
+extension ValidateSaasPremiumAvailableExtension on MailboxDashBoardController {
+  bool get isPremiumAvailable {
+    if (accountId.value == null || sessionCurrent == null) {
+      return false;
+    }
+
+    final saasAccount = sessionCurrent?.getSaaSAccountCapability(
+      accountId.value!,
+    );
+
+    return saasAccount?.isPremiumAvailable ?? false;
+  }
+
+  void loadPaywallUrl() {
+    final getPaywallUrlInteractor = getBinding<GetPaywallUrlInteractor>();
+    final jmapUrl = dynamicUrlInterceptors.jmapUrl;
+
+    if (getPaywallUrlInteractor != null && jmapUrl != null) {
+      consumeState(getPaywallUrlInteractor.execute(jmapUrl));
+    } else {
+      paywallUrlPattern = null;
+    }
+  }
+
+  void loadPaywallUrlSuccess(PaywallUrlPattern newPattern) {
+    paywallUrlPattern = newPattern;
+  }
+
+  void navigateToPaywall(BuildContext context) {
+    if (paywallUrlPattern == null) {
```

Comment:
how about we should get again the paywall when user try to click on it?

Reply:
We will add action for toast message. See below demo:


https://github.com/user-attachments/assets/c4994c71-b701-4d1d-bce5-00a0ec92addd